components/menu/menu.tsx

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {ReactNode, useState, MouseEvent} from 'react';
+import {Menu, MenuList} from '@mui/material';
+
+import Tooltip from 'components/tooltip';
+import CompassDesignProvider from 'components/compass_design_provider';
+import OverlayTrigger from 'components/overlay_trigger';
+
+import type {PropsFromRedux} from './index';
+
+const OVERLAY_TIME_DELAY = 500;
+
+interface Props extends PropsFromRedux {
+
+    // Anchor button props
+    anchorId?: string;
+    anchorNode?: ReactNode;
+    anchorClassName?: string;
+    anchorAriaLabel?: string;
+
+    // Menu props
+    menuId?: string;
+    menuAriaLabel?: string;
+
+    // Tooltip props
+    tooltipPlacement?: 'top' | 'bottom' | 'left' | 'right';
+    tooltipId?: string;
+    tooltipText?: string;
+    tooltipClassName?: string;
+
+    children: ReactNode;
+}
+
+export function MenuComponent(props: Props) {
+    const [anchorElement, setAnchorElement] = useState<null | HTMLElement>(null);
+    const isMenuOpen = Boolean(anchorElement);
+
+    const handleMenuOpen = (event: MouseEvent<HTMLButtonElement>) => {
+        event.preventDefault();
+        setAnchorElement(event.currentTarget);
+    };
+
+    const handleMenuClose = (event: MouseEvent<HTMLDivElement | HTMLUListElement>) => {
+        event.preventDefault();
+        setAnchorElement(null);
+    };
+
+    return (
+        <CompassDesignProvider theme={props.theme}>
+            <OverlayTrigger
+                delayShow={OVERLAY_TIME_DELAY}
+                placement={props?.tooltipPlacement ?? 'top'}
+                overlay={
+                    <Tooltip
+                        id={props.tooltipId}
+                        className={props.tooltipClassName}
+                    >
+                        {props.tooltipText}
+                    </Tooltip>
+                }
+                disabled={!props.tooltipText || isMenuOpen}
```

Comment:
0/5, but it might make sense to just not use an `OverlayTrigger` at all if `tooltipText` isn't provided

Reply:
✅  Updated as per request/discussion